analyzers/src/SonarAnalyzer.CSharp/Rules/InsteadOfAny.cs

```diff
@@ -21,24 +21,20 @@
 namespace SonarAnalyzer.Rules.CSharp;
 
 [DiagnosticAnalyzer(LanguageNames.CSharp)]
-public sealed class ExistsInsteadOfAny : ExistsInsteadOfAnyBase<SyntaxKind, InvocationExpressionSyntax>
+public sealed class InsteadOfAny : InsteadOfAnyBase<SyntaxKind, InvocationExpressionSyntax>
 {
     protected override ILanguageFacade<SyntaxKind> Language => CSharpFacade.Instance;
 
-    protected override bool HasOneArgument(InvocationExpressionSyntax node) =>
-        node.HasExactlyNArguments(1);
-
-    protected override bool IsValueEquality(InvocationExpressionSyntax node, SemanticModel model) =>
-        node.ArgumentList.Arguments[0].Expression is SimpleLambdaExpressionSyntax lambda
+    protected override bool IsSimpleEqualityCheck(InvocationExpressionSyntax node, SemanticModel model) =>
+        GetArgumentExpression(node, 0) is SimpleLambdaExpressionSyntax lambda
         && lambda.Parameter.Identifier.ValueText is var lambdaVariableName
         && lambda.Body switch
         {
             BinaryExpressionSyntax binary =>
                 binary.OperatorToken.IsAnyKind(SyntaxKind.EqualsEqualsToken)
                 && HasBinaryValidOperands(lambdaVariableName, binary.Left, binary.Right, model),
```

Comment:
Nitpick: I think a slightly better name is `HasValidBinaryOperands`, since you are looking for valid (binary operands), not binary (valid operands). 

Same in VB.NET

Reply:
We already have one method named `AreValidOperands`. I'm afraid it will be confusing, wdyt?

Reply:
I think it's more correct and  readable in the order `valid (modifier thing)` instead of `(modifier) valid (thing)`, but It's okay to leave it as is as well. Feel free to implement this or not and resolve the issue. Same with the other similar one :)